File: shared/src/androidMain/kotlin/app/inspiry/core/manager/ToastManagerImpl.kt
package app.inspiry.core.manager

import android.content.Context
import android.widget.Toast

class ToastManagerImpl(val context: Context) : ToastManager {

    override fun displayToast(text: String, length: ToastLength) {

        Toast.makeText(
            context, text, when (length) {
                ToastLength.SHORT -> Toast.LENGTH_SHORT
                ToastLength.LONG -> Toast.LENGTH_LONG
            }
        ).show()
    }
}

File: iosApp/iosApp/project/manager/ToastManagerImpl.swift
//
//  ToastManagerImpl.swift
//  iosApp
//
//  Created by vlad on 10/12/21.
//

import Foundation
import Toaster
import shared

class ToastManagerImpl: ToastManager {
    func displayToast(text: String, length: ToastLength) {
        Toast(text: text, duration: length == ToastLength.short_ ? Delay.short : Delay.long).show()
    }
}
